packages/circuits/test/circuits/initial.test.circom

```diff
@@ -0,0 +1,5 @@
+pragma circom 2.0.0;
+
+include "../../src/initial.circom";
+
+component main {public [deckRoot, newDeckRoot, initialDeckTailCardIndex, handRoot, newHandRoot, drawnCardIndices]} = Initial(4, 2);
```

Comment:
Any insight on the performance of this? What about bigger numbers? (I'd be particularly interested in 6 levels (64 cards) and a hand of 7 cards or so which mimics the deck/hand sizes in popular card games.)

Reply:
Good point, let me try running some different values and compare the performance.

Reply:
Okay tried it out! Proof generation takes around 22-25 seconds averagely for 6 levels and 7 cards. More detailed constraint numbers are in comment below.

Reply:
Nice work! But damn, that's way too much for the game :D

I'm guessing the shuffling idea Brian suggested is the way to go: use a proof of shuffle, then we can have the contract compute all random (unique!) draw indices and pass them into the initial hand proof and use the bitfield approach to reduce to a single pass over the deck and a single pass over the hand.

This will require changing the contracts so that there the randomness seed for the initial hand is only known after the shuffle is committed to.